src/main/java/org/broadinstitute/hellbender/tools/walkers/rnaseq/SplitNCigarReads.java

```diff
@@ -91,46 +90,43 @@
-     */
-    protected OverhangFixingManager overhangManager;
+    private SAMFileGATKReadWriter outputWriter;
+    private OverhangFixingManager overhangManager;
+    ReadTransformer rnaReadTransform;
 
     @Override
-    protected Object doWork() {
-        IOUtil.assertFileIsReadable(INPUT);
-        final SamReader in = SamReaderFactory.makeDefault().open(INPUT);
-        final Iterable<GATKRead> readIter = new SAMRecordToReadIterator(in.iterator());
-        final SAMFileGATKReadWriter outputWriter = initialize(in);
-
-        final ReadTransformer rnaReadTransform = REFACTOR_NDN_CIGAR_READS ? new NDNCigarReadTransformer() : ReadTransformer.identity();
-
-        StreamSupport.stream(readIter.spliterator(), false)
-                .map(rnaReadTransform)
-                .forEach(read -> splitNCigarRead(read, overhangManager));
-        overhangManager.close();
-        CloserUtil.close(in);
-        CloserUtil.close(outputWriter);
-        return null;
+    public CountingReadFilter makeReadFilter() {
+        return new CountingReadFilter("ALL_READS", ReadFilterLibrary.ALLOW_ALL_READS);
     }
-
-    private SAMFileGATKReadWriter initialize(final SamReader in) {
-        final SAMFileHeader outputHeader = ReadUtils.cloneSAMFileHeader(in.getFileHeader());
-        //preSorted is set to false for this writer since ReadCoordinateComparator doesn't match HTSJDK's when an unmapped read has a mapped mate.
-        //Also, we don't want to assume that reads will be written in order by the manager because in deep, deep pileups it won't work
-        final SAMFileGATKReadWriter outputWriter = new SAMFileGATKReadWriter(ReadUtils.createCommonSAMWriter(OUTPUT, REFERENCE_SEQUENCE, outputHeader, false, false, false));
-
+    
+    @Override
+    public void onTraversalStart() {
+        final IndexedFastaSequenceFile referenceReader;
+        SAMFileHeader header = getHeaderForSAMWriter();
+        rnaReadTransform = REFACTOR_NDN_CIGAR_READS ? new NDNCigarReadTransformer() : ReadTransformer.identity();
         try {
-            final IndexedFastaSequenceFile referenceReader = new CachingIndexedFastaSequenceFile(REFERENCE_SEQUENCE);
-            GenomeLocParser genomeLocParser= new GenomeLocParser(referenceReader.getSequenceDictionary());
-            overhangManager = new OverhangFixingManager(outputHeader, outputWriter, genomeLocParser, referenceReader, MAX_RECORDS_IN_MEMORY, MAX_MISMATCHES_IN_OVERHANG, MAX_BASES_TO_CLIP, doNotFixOverhangs);
-            return outputWriter;
+            referenceReader = new CachingIndexedFastaSequenceFile(referenceArguments.getReferenceFile());
+            GenomeLocParser genomeLocParser = new GenomeLocParser(getBestAvailableSequenceDictionary());
+            outputWriter = new SAMFileGATKReadWriter(ReadUtils.createCommonSAMWriter(OUTPUT, referenceArguments.getReferenceFile(), header, false, false, false));
```

Comment:
So `ReadWalker` has a method "createSAMWriter" that's probably better to use than creating one manually.